src/main/java/org/folio/rest/impl/PatronBlockConditionsAPI.java

```diff
@@ -0,0 +1,133 @@
+  }
+
+  @Validate
+  @Override
+  public void postPatronBlockConditions(String lang, PatronBlockCondition entity,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    Errors errors = isEntityValid(entity, asyncResultHandler);
+    if (errors != null) {
+      asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
+        PatronBlockConditions.PostPatronBlockConditionsResponse
+          .respond422WithApplicationJson(errors)));
+    }
+
+    PgUtil.post(PATRON_BLOCK_CONDITIONS, entity, okapiHeaders, vertxContext,
+      PostPatronBlockConditionsResponse.class, asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void putPatronBlockConditionsByPatronBlockConditionId(
+    String patronBlockConditionId, String lang, PatronBlockCondition entity,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    Errors errors = isEntityValid(entity, asyncResultHandler);
+    if (errors != null) {
+      asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
+        PatronBlockConditions.PutPatronBlockConditionsByPatronBlockConditionIdResponse
+          .respond422WithApplicationJson(errors)));
+    }
+
+    PgUtil.put(PATRON_BLOCK_CONDITIONS, entity, patronBlockConditionId, okapiHeaders,
+      vertxContext, PutPatronBlockConditionsByPatronBlockConditionIdResponse.class, asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void getPatronBlockConditionsByPatronBlockConditionId(
+    String patronBlockConditionId, String lang, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    PgUtil.getById(PATRON_BLOCK_CONDITIONS, PatronBlockCondition.class, patronBlockConditionId,
+      okapiHeaders, vertxContext, GetPatronBlockConditionsByPatronBlockConditionIdResponse.class,
+      asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void deletePatronBlockConditionsByPatronBlockConditionId(
+    String patronBlockConditionId, String lang, Map<String,
+    String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    PgUtil.deleteById(PATRON_BLOCK_CONDITIONS, patronBlockConditionId, okapiHeaders,
+      vertxContext, DeletePatronBlockConditionsByPatronBlockConditionIdResponse.class,
+      asyncResultHandler);
+  }
+
+  private Errors isEntityValid(PatronBlockCondition entity, Handler<AsyncResult<Response>> asyncResultHandler) {
+    Errors errors = null;
+    if (!isEntityMessageValid(entity)) {
+      errors = createValidationErrorMessage("message", entity.getId(),
+        "Message to be displayed is a required field if one or more blocked actions selected");
+      asyncResultHandler.handle(Future.succeededFuture(
+        Proxiesfor.PostProxiesforResponse.respond422WithApplicationJson(errors)));
+    }
```

Comment:
Can you return at the end of the if clause to avoid executing the second if clause after the first if clause has been triggered?

    if (...) {
      Errors errors = ...
      ...
      return errors;
    }
    if (...) {
      Errors errors = ...
      ...
      return errors;
    }
    return null;

Reply:
Or use `return createValidationErrorMessage(...);` when removing `asyncResultHandler`.

Reply:
thanks for the help, 
removed